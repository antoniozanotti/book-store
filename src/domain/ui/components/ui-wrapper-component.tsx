import { UiWrapperComponentInterface } from "./ui-wrapper-component-interface";

export function UiWrapperComponent({ children, ...props }: UiWrapperComponentInterface) {
  const divStyles = "w-full mx-auto xl:max-w-screen-xl 2xl:max-w-screen-2xl";
  props.className = props.className
    ? `${divStyles} ${props.className}`
    : divStyles;

  return <div {...props}>{children}</div>;
}
