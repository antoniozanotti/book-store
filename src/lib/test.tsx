import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
