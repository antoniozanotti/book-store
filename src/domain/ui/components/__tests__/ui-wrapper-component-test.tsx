import { render, screen } from "@/lib/test";
import { UiWrapperComponent } from "../ui-wrapper-component";

describe("UiWrapperComponent", () => {
  it("should render", async () => {
    const { container } = render(<UiWrapperComponent>Test</UiWrapperComponent>);
    expect(container).toMatchSnapshot();
  });
});