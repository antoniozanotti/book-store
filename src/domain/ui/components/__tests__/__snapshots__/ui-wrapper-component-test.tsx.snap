// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UiWrapperComponent should render 1`] = `
<div>
  <div
    data-testid="container"
  >
    <div
      class="w-full mx-auto xl:max-w-screen-xl 2xl:max-w-screen-2xl"
    >
      Test
    </div>
  </div>
</div>
`;
